src/lib/Server/Input/Parser/Criterion/Image.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Rest\Server\Input\Parser\Criterion;
+
+use Ibexa\Contracts\Core\Repository\Values\Content\Query\Criterion\Image as ImageCriterion;
+use Ibexa\Contracts\Rest\Input\ParsingDispatcher;
+use Ibexa\Rest\Input\BaseParser;
+use Ibexa\Rest\Server\Exceptions\ValidationFailedException;
+use Ibexa\Rest\Server\Validation\Builder\Input\Parser\Criterion\ImageCriterionValidatorBuilder;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+final class Image extends BaseParser
+{
+    public const IMAGE_CRITERION = 'ImageCriterion';
+    public const FIELD_DEF_IDENTIFIER_KEY = 'fieldDefIdentifier';
+    public const MIME_TYPES_KEY = 'mimeTypes';
+
+    private ValidatorInterface $validator;
+
+    public function __construct(ValidatorInterface $validator)
+    {
+        $this->validator = $validator;
+    }
+
+    /**
+     * @param array<mixed> $data
+     *
+     * @return \Ibexa\Contracts\Core\Repository\Values\Content\Query\Criterion\Image
+     *
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\InvalidArgumentException
+     * @throws \Ibexa\Contracts\Core\Repository\Exceptions\InvalidCriterionArgumentException
+     */
+    public function parse(array $data, ParsingDispatcher $parsingDispatcher): ImageCriterion
+    {
+        $this->validateInputArray($data);
+
+        $criterionData = $data[self::IMAGE_CRITERION];
+
+        return new ImageCriterion(
+            $criterionData[self::FIELD_DEF_IDENTIFIER_KEY],
+            $this->extractImageCriteria($criterionData)
+        );
+    }
+
+    /**
+     * @param array<mixed> $data
```

Comment:
> [!NOTE]  
> Optional.

You can use `@phpstan-assert` to inform static analysis about the expected shape of data after validation.

See https://phpstan.org/writing-php-code/narrowing-types#custom-type-checking-functions-and-methods for details.

Reply:
As we discussed with the dev team, tests coverage will be added later.